ApplicationLibCode/ProjectDataModel/Summary/RimEnsembleStatisticsCase.cpp

```diff
@@ -143,41 +140,44 @@ void RimEnsembleStatisticsCase::calculate( const std::vector<RimSummaryCase*>& s
     auto [minTimeStep, maxTimeStep]   = findMinMaxTimeStep( sumCases, inputAddress );
     RiaDefines::DateTimePeriod period = findBestResamplingPeriod( minTimeStep, maxTimeStep );
 
-    std::vector<time_t>              allTimeSteps;
-    std::vector<std::vector<double>> caseAndTimeStepValues;
-    caseAndTimeStepValues.reserve( sumCases.size() );
+    RiaTimeHistoryCurveMerger curveMerger;
+
     for ( const auto& sumCase : sumCases )
     {
         const auto& reader = sumCase->summaryReader();
         if ( reader )
         {
             const std::vector<time_t>& timeSteps = reader->timeSteps( inputAddress );
-            auto [isOk, values]                  = reader->values( inputAddress );
+            const auto [isOk, values]            = reader->values( inputAddress );
 
             if ( values.empty() ) continue;
 
-            if ( !includeIncompleteCurves && timeSteps.size() != values.size() ) continue;
+            if ( !includeIncompleteCurves && ( timeSteps.back() != maxTimeStep ) ) continue;
```

Comment:
Not sure about this check? Could be different eventhough the last step is the same?

Reply:
timeSteps.back() access is also unchecked? Could be empty?

Reply:
The number of time steps in an ensemble can vary. The last time step is identical for all completed realizations. I added a threshold to make the computation even more robust.